packages/dds/merge-tree/src/partialLengths.ts

```diff
@@ -242,21 +293,17 @@ export class PartialSequenceLengths {
                 if (seqLTE(segment.seq, collabWindow.minSeq)) {
                     combinedPartialLengths.minLength += segment.cachedLength;
                 } else {
-                    if (segment.seq !== UnassignedSequenceNumber) {
-                        PartialSequenceLengths.insertSegment(combinedPartialLengths, segment);
-                    }
+                    PartialSequenceLengths.insertSegment(combinedPartialLengths, segment);
                 }
                 const removalInfo = toRemovalInfo(segment);
                 if (seqLTE(removalInfo?.removedSeq, collabWindow.minSeq)) {
                     combinedPartialLengths.minLength -= segment.cachedLength;
-                } else {
-                    if (removalInfo !== undefined
-                        && removalInfo.removedSeq !== UnassignedSequenceNumber) {
-                        PartialSequenceLengths.insertSegment(
-                            combinedPartialLengths,
-                            segment,
-                            removalInfo);
-                    }
+                } else if (removalInfo !== undefined ||
+                    (segment.localRemovedSeq !== undefined && segment.localRemovedSeq !== UnassignedSequenceNumber)) {
+                    PartialSequenceLengths.insertSegment(
+                        combinedPartialLengths,
+                        segment,
+                        segment as IRemovalInfo);
```

Comment:
```suggestion
                        removalInfo);
```

Reply:
oops, I meant to revisit this. Was a temporary kludge to make sure `localRemovedSeq` got included on the removal info object. I think we should probably just make `IRemovalInfo` include that field when it's defined on `segment`. Need to do a short audit though to make sure that doesn't mess with anything.